zcbor 0.8.1 (#13768)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

    * The :kconfig:option:`CONFIG_DFU_TARGET_FULL_MODEM_USE_EXT_PARTITION` Kconfig option to be automatically enabled when ``nordic,pm-ext-flash`` is chosen in the devicetree.
      See :ref:`partition_manager` for details.

>     * Adapt to API changes in zcbor 0.8.x.

Reviewer: ```suggestion
    * Adapted to API changes in zcbor 0.8.x.
```

* :ref:`lib_fmfu_fdev` library:

  * Updated:

>     * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.

Reviewer: ```suggestion
    * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.
```

  * Fixed a potential race condition, where two threads might corrupt a shared response buffer.

* :ref:`lib_modem_attest_token` library:

>   * Adapt to API changes in zcbor 0.8.x.

Reviewer: ```suggestion
  * Updated to adapt to API changes in zcbor 0.8.x.
```

      This parameter determines whether CoAP CON or NON messages are used.

>     * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.

Reviewer: ```suggestion
    * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.
```

  * Updated the device mode callback to send an event when the provisioning state changes.
  * Fixed file descriptor handling by setting the :c:struct:`coap_client` structure's ``fd`` field to ``-1`` when closing the socket.
  * Added the :kconfig:option:`CONFIG_NRF_PROVISIONING_PRINT_ATTESTATION_TOKEN` option to enable printing the attestation token when the device is not yet claimed.

>   * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.

Reviewer: ```suggestion
  * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.
```

>   * Updated the device mode callback to send an event when the provisioning state changes.
>   * Fixed file descriptor handling by setting the :c:struct:`coap_client` structure's ``fd`` field to ``-1`` when closing the socket.
>   * Added the :kconfig:option:`CONFIG_NRF_PROVISIONING_PRINT_ATTESTATION_TOKEN` option to enable printing the attestation token when the device is not yet claimed.
>   * Regenerate cbor code with zcbor 0.8.1 and adapt to API changes in zcbor 0.8.x.

Reviewer: ```suggestion
  * Added the :kconfig:option:`CONFIG_NRF_PROVISIONING_PRINT_ATTESTATION_TOKEN` option to enable printing the attestation token when the device is not yet claimed.
  * Updated:
  
    * The device mode callback sends an event when the provisioning state changes.
    * Regenerated cbor code with zcbor 0.8.1 and adapted to API changes in zcbor 0.8.x.

  * Fixed file descriptor handling by setting the :c:struct:`coap_client` structure's ``fd`` field to ``-1`` when closing the socket.
```

Author: Your last proposal changes text unrelated to my changes, can you confirm that I should change it here?

Reviewer: It is just grouping related points together. I can fix this later if needed.

Author: No worries, I added it, just wanted to confirm.

zcbor
=====

zcbor has been updated from 0.7.0 to 0.8.1.

> Full release notes can be found at:
> https://github.com/zephyrproject-rtos/zcbor/blob/0.8.0/RELEASE_NOTES.md and
> https://github.com/zephyrproject-rtos/zcbor/blob/0.8.1/RELEASE_NOTES.md

Reviewer: Please add the entries : 
`zcbor 0.8.0 release notes`: https://github.com/zephyrproject-rtos/zcbor/blob/0.8.0/RELEASE_NOTES.md 
`zcbor 0.8.0 release notes`: https://github.com/zephyrproject-rtos/zcbor/blob/0.8.1/RELEASE_NOTES.md 
to doc/nrf/links.txt for the links to work.
```suggestion
For more information see the `zcbor 0.8.0 release notes`_ and the `zcbor 0.8.1 release notes`_.
```

> Highlights:
> 
> * Add support for unordered maps
> * Performance improvements
> * Naming improvements for generated code
> * Bugfixes

Reviewer: ```suggestion
* Added:

  * Support for unordered maps.
  * Performance improvements.
  * Naming improvements for generated code.
  * Bugfixes.
```
